backend/my_executors.py

```diff
@@ -90,109 +92,32 @@ def preprocessing(self, sequences: Sequence[str]) -> List[str]:
 
 
 class MyIndexer(Executor):
-    """Indexer class"""
+    """Indexer Executor"""
 
     def __init__(self, **kwargs):
+        log("Initialising Indexer.")
         super().__init__(**kwargs)
-        self._docs = DocumentArray()
-        if os.path.exists(self.save_path):
-            self._docs = DocumentArray.load(self.save_path)
+        if os.path.exists(embeddings_path):
+            self._docs = DocumentArray().load(embeddings_path)
         else:
             self._docs = DocumentArray()
 
+        log(f"Loaded {len(self._docs)} proteins with embeddings.")
+
     @requests(on="/index")
     def index(self, docs: DocumentArray, **kwargs):
         self._docs.extend(docs)
+        log("Saving embeddings.")
         self.save()
         return docs
 
     @requests(on="/search")
     def search(self, docs: "DocumentArray", **kwargs):
-        # We preload `_docs` in the `__init__`. This might need changes for testing.
-        proteins = self._docs #DocumentArray.load(self.save_path)
-        results = DocumentArray()
-
-        """Search method and called methods are as in the chatbot example"""
-        a = np.stack(docs.get_attributes("embedding"))
-        b = np.stack(proteins.get_attributes("embedding"))
-        q_emb = _ext_A(_norm(a))
-        d_emb = _ext_B(_norm(b))
-        dists = _cosine(q_emb, d_emb)
-        idx, dist = self._get_sorted_top_k(dists, top_k)
-        for _q, _ids, _dists in zip(docs, idx, dist):
-            for _id, _dist in zip(_ids, _dists):
-                d = Document(self._docs[int(_id)], copy=True)
-                d.scores["cosine"] = 1 - _dist
-                _q.matches.append(d)
-                results.append(d)
-
-        self.save_results(results)
-
-        return results
-
-    @staticmethod
-    def _get_sorted_top_k(
-        dist: "np.array", top_k: int
-    ) -> Tuple["np.ndarray", "np.ndarray"]:
-        if top_k >= dist.shape[1]:
-            idx = dist.argsort(axis=1)[:, :top_k]
-            dist = np.take_along_axis(dist, idx, axis=1)
-        else:
-            idx_ps = dist.argpartition(kth=top_k, axis=1)[:, :top_k]
-            dist = np.take_along_axis(dist, idx_ps, axis=1)
-            idx_fs = dist.argsort(axis=1)
-            idx = np.take_along_axis(idx_ps, idx_fs, axis=1)
-            dist = np.take_along_axis(dist, idx_fs, axis=1)
-
-        return idx, dist
+        log(f'Computing metric to {len(self._docs)} proteins.')
 
-    @property
-    def save_path(self):
-        directory = "".join(embeddings_path.split("/")[0:-1])
-        os.makedirs(directory, exist_ok=True)
-
-        return embeddings_path
+        docs.match(self._docs, metric='cosine', limit=top_k)
```

Comment:
At a certain point, the `top_k` argument should be given as input in the request. We can also parametrize it in the frontend.